### ISSUE TITLE
Uploading build fails with EOF

### ISSUE THREAD

**Issue by paulkernfeld** [convox/rack#2223]
I've only ever seen this once, so it may be nondeterministic.

```
$ convox deploy --app itinerary-getter
Deploying itinerary-getter
Creating tarball... OK
Uploading: 68.28 MB / 68.28 MB [========================] 100.00 % 25s
Starting build... ERROR: Post https://uat-convox-561984580.us-west-2.elb.amazonaws.com/apps/itinerary-getter/builds: EOF
```

**Comment by paulkernfeld** [comment]
Oh no! This happened again! This is occurring from my CI platform, Codeship.

```
Creating tarball... OK
Uploading: 74.12 MB / 74.12 MB [=========================] 100.00 % 6s
Starting build... ERROR: Post https://uat-convox-561984580.us-west-2.elb.amazonaws.com/apps/booker/builds: EOF
```

**Comment by matthewford** [comment]
I've seen this a few times too, depends on your net connection.

**Comment by paulkernfeld** [comment]
Yep, that's what I had suspected. It's probably not safe to assume that users will have a reliable network connection, so it would be cool if Convox automatically retried or resumed in this case.

**Comment by paulkernfeld** [comment]
Hmm, I recently got a similar error:

```
Promoting RBUWGQDWKWN... ERROR: Post https://prod-convox-336862854.us-west-2.elb.amazonaws.com/apps/booker/releases/RBUWGQDWKWN/promote: EOF
```

**Comment by mathomas** [comment]
Just thought I'd add here that we had a similar thing on `convox deploy`.  About 4 of 5 deploys would fail.  I did a bunch of research and found that the convox/api container in the ECS instance was actually exiting.  No useful error message, just dumped out of the container.

After some research in CloudWatch I built a theory that the task was running out of memory.  So I increased the memory allocated to the task using the command line:

```
-bash-4.2$ convox rack params
NAME                     VALUE
Agent                    Yes
Ami
ApiCount                 2
ApiCpu                   128
ApiMemory                128  <==
Autoscale                No
```
Increased by 50%:

```
-bash-4.2$ convox rack params set ApiMemory=192
Updating parameters… OK
```

After this update we had four good builds in a row.  Not sure this will apply to any other team's situation, but it seemed to address our problem.

**Comment by paulkernfeld** [comment]
@mathomas thank you for commenting! I had made some progress in the mean time but forgot to comment here. I actually came to the same solution as you did, so this fix will probably helpful for others as well.

One specific thing that I noticed when running `convox rack logs` was that it actually mentioned that the convox/api container was killed due to OOM.

In order to fix another of these errors, I recall needing to increase the amount of memory allocated to builds (also in `rack params`).
